### ISSUE TITLE
UndefinedFunction with <extraFiles>

### ISSUE THREAD

**Issue by ro0NL** [vimeo/psalm#9884]
This seems related to #5791 

I have this file structure:

```
# src/file.php
<?php foo();

# src/inc/file.php
function foo() { return 1; }
```

And this psalm config:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<psalm
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config"
    cacheDirectory="var"
>
    <projectFiles>
        <directory name="src"/>
    </projectFiles>
    <extraFiles>
        <directory name="src/inc"/>
        <file name="src/inc/file.php"/>
    </extraFiles>
</psalm>
```

Running psalm produces unexpected error:

```
$ psalm        
Warning: "findUnusedBaselineEntry" will be defaulted to "true" in Psalm 6. You should explicitly enable or disable this setting.
Warning: "findUnusedCode" will be defaulted to "true" in Psalm 6. You should explicitly enable or disable this setting.
Warning: "findUnusedBaselineEntry" will be defaulted to "true" in Psalm 6. You should explicitly enable or disable this setting.
Warning: "findUnusedCode" will be defaulted to "true" in Psalm 6. You should explicitly enable or disable this setting.
Target PHP version: 8.1 (inferred from current PHP version).
Scanning files...
Analyzing files...

E░

ERROR: UndefinedFunction - src/file.php:3:1 - Function foo does not exist (see https://psalm.dev/021)
foo()


------------------------------
1 errors found
------------------------------

Checks took 1.29 seconds and used 176.077MB of memory
Psalm was unable to infer types in the codebase

$ psalm -v
Psalm 5.12.0@f90118cdeacd0088e7215e64c0c99ceca819e176
```

It works with `<stubs><file name="src/inc/file.php"/></stubs>` but this does not scale. AFAIK directories are not supported. Also, the file is not a stub but actual code.

The reason there's no `require` in code is because in runtime files are requried at higher level already, and i don't want to touch it.

**Comment by orklah** [comment]
Hey @ro0NL sorry for not answering before.

Did you find a solution on your own for this issue?

**Comment by ro0NL** [comment]
@orklah added everything to `<stubs>` :)
